zerver/views/webhooks/hellosign.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from six import text_type
+from six.moves import range
+from typing import Dict, Any, Optional, Union
+
+def format_body(signatories):
+    # type: (List[Dict[str, Any]]) -> Any
+    RESULT_TEMPLATE = "The {contract_title}"
+    for i in range(len(signatories)):
+        signatory = signatories[i]
+        identity = str(i) # type: str
+        if signatory['status_code'] == 'awaiting_signature':
+            RESULT_TEMPLATE += " is awaiting the signature of {name_" + identity + "}"
+        else:
+            RESULT_TEMPLATE += " was just {status_" + identity + "} by {name_" + identity + "}"
+
+        if i+1 == len(signatories):
+            RESULT_TEMPLATE += '.'
+        elif i+2 == len(signatories):
+            RESULT_TEMPLATE += ' and'
+        elif i+3 != len(signatories):
+            RESULT_TEMPLATE += ','
+    return RESULT_TEMPLATE
+
+def ready_payload(signatories, payload):
+    # type: (List[Dict[str, Any]], Dict[str, Dict[str, Any]]) -> None
+    for i in range(len(signatories)):
+        signatory = signatories[i]
+        identity = str(i)
+        payload['name_' + identity] = signatory['signer_name']
+        payload['status_' + identity] = signatory['status_code']
+    return
+
+@api_key_only_webhook_view('HelloSign')
+@has_request_variables
+def api_hellosign_webhook(request, user_profile, client,
+                          payload=REQ(argument_type='body'),
+                          stream=REQ(default='hellosign'),
+                          topic=REQ(default=None)):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Dict[str, Any]], text_type, text_type) -> HttpResponse
+    try:
+        model_payload = {'contract_title': payload['signature_request']['title']}
+        ready_payload(payload['signature_request']['signatures'], model_payload)
```

Comment:
It would be nice to create this method in a functional way (not modifying the object by reference).